### PR TITLE
Area2D Documentation + constants in _bind_methods

### PR DESCRIPTION
**Pull request by willnationsdev**
Please let me know if I should update anything.

### CODE REVIEW

File: doc/base/classes.xml [hunk 1]
```diff
@@ -5395,33 +5395,52 @@
...
+      The rate at which objects stop spinning in this area, excluding external forces' influences, measured as a fraction of speed lost per second (1.0 - stops immediately, 0.0 - never stops).
+			The period between physics calculations affects this rate (e.g. 1.0 - only stops movement after 1 second when physics is evaluated every 1 second). 
 		</member>
-		<member name="audio_bus_name" type="String" setter="set_audio_bus" getter="get_audio_bus" brief="">
+		<member name="audio_bus_name" type="String" setter="set_audio_bus" getter="get_audio_bus" brief="name of audio bus">
+      The name of the Area2D's audio bus.
 		</member>
-		<member name="audio_bus_override" type="bool" setter="set_audio_bus_override" getter="is_overriding_audio_bus" brief="">
+		<member name="audio_bus_override" type="bool" setter="set_audio_bus_override" getter="is_overriding_audio_bus" brief="Whether to override audio bus">
+      Whether to override the default audio bus with the Area2D's.
```
Comment (cbscribe) [inline doc/base/classes.xml]: Boolean members should follow styleguide: https://github.com/GDquest/godot-3-guides/blob/master/contribution-guides/class-reference-styleguide.md#use-if-true-to-describe-booleans

File: doc/base/classes.xml [hunk 2]
```diff
@@ -5395,33 +5395,52 @@
...
+			This value multiplies the gravity vector, whether it is the given vector ([method set_gravity_vector]), or a calculated one (when using a center of gravity).
 		</member>
-		<member name="gravity_distance_scale" type="float" setter="set_gravity_distance_scale" getter="get_gravity_distance_scale" brief="">
+		<member name="gravity_distance_scale" type="float" setter="set_gravity_distance_scale" getter="get_gravity_distance_scale" brief="gravity drop rate">
+      The falloff factor for point gravity. The greater this value is, the faster the strength of gravity decreases with the square of distance.
 		</member>
-		<member name="gravity_point" type="bool" setter="set_gravity_is_point" getter="is_gravity_a_point" brief="">
+		<member name="gravity_point" type="bool" setter="set_gravity_is_point" getter="is_gravity_a_point" brief="is gravity a point?">
+      Whether gravity should be calculated from a particular point during a space override (see [method set_space_override_mode]). If a point, [Vector2] position is set with [method set_gravity_vector].
```
Comment (cbscribe) [inline doc/base/classes.xml]: Change per styleguide

File: doc/base/classes.xml [hunk 3]
```diff
@@ -5395,33 +5395,52 @@
...
+      The rate at which objects stop moving in this area, excluding external forces' influences, measured as a fraction of speed lost per second (1.0 - stops immediately, 0.0 - never stops).
+			The period between physics calculations affects this rate (e.g. 1.0 - only stops movement after 1 second when physics is evaluated every 1 second). 
 		</member>
-		<member name="monitorable" type="bool" setter="set_monitorable" getter="is_monitorable" brief="">
+		<member name="monitorable" type="bool" setter="set_monitorable" getter="is_monitorable" brief="others cannot detect this?">
+      Whether other monitoring areas can detect this Area2D (is it undetectable at the moment?).
 		</member>
-		<member name="monitoring" type="bool" setter="set_monitoring" getter="is_monitoring" brief="">
+		<member name="monitoring" type="bool" setter="set_monitoring" getter="is_monitoring" brief="this cannot detect others?">
+      Whether this Area2D detects bodies/areas entering/exiting it (can it detect others at the moment?).
```
Comment (cbscribe) [inline doc/base/classes.xml]: Change per styleguide

File: doc/base/classes.xml [hunk 4]
```diff
@@ -5083,10 +5083,10 @@
 </class>
 <class name="Area2D" inherits="CollisionObject2D" category="Core">
 	<brief_description>
-		General purpose area detection and influence for 2D physics.
+	  General purpose area detection for 2D physics. Can override 2D physics properties within range.
```
Comment (NathanLovato) [inline doc/base/classes.xml]: 2 leading spaces to remove

File: doc/base/classes.xml [hunk 5]
```diff
@@ -5395,33 +5395,52 @@
 		</method>
 	</methods>
 	<members>
-		<member name="angular_damp" type="float" setter="set_angular_damp" getter="get_angular_damp" brief="">
+		<member name="angular_damp" type="float" setter="set_angular_damp" getter="get_angular_damp" brief="rate of spin loss">
+      The rate at which objects stop spinning in this area, excluding external forces' influences, measured as a fraction of speed lost per second (1.0 - stops immediately, 0.0 - never stops).
```
Comment (NathanLovato) [inline doc/base/classes.xml]: You have spaces for indent on all member vars description. When you look at the git diff, before you commit, you shouldn't see the `<member` lines in red. Select the whole <members> section and convert the space indents to tabs. You may also want to squash all commits with:
`git rebase -i HEAD~4`

The description is also quite complex. Try to simplify it and chop it up into shorter sentences. Avoid parentheses, as they lengthen sentences. Example:
`The rate at which objects stop spinning in this area. 'angular_damp' represents the amount of speed lost per second. If set to 1.0, physics bodies in the area stop rotating instantly. At 0.0 they never slow down. Doesn't take external forces in account.`

Also avoid passive voice.

File: doc/base/classes.xml [hunk 6]
```diff
@@ -5083,10 +5083,10 @@
 </class>
 <class name="Area2D" inherits="CollisionObject2D" category="Core">
 	<brief_description>
-		General purpose area detection and influence for 2D physics.
+	  General purpose area detection for 2D physics. Can override 2D physics properties within range.
 	</brief_description>
 	<description>
-		General purpose area detection for 2D physics. Areas can be used for detection of objects that enter/exit them, as well as overriding space parameters (changing gravity, damping, etc). For this, use any space override different from AREA_SPACE_OVERRIDE_DISABLE and point gravity at the center of mass.
+		General purpose area detection for 2D physics (see [CollisionObject2D] for usage). Areas can detect objects that enter or exit them. They can also override space parameters such as gravity, damping, etc). For this, set the 'space_override' property to any value other than AREA_SPACE_OVERRIDE_DISABLE and point gravity at the center of mass.
```
Comment (NathanLovato) [inline doc/base/classes.xml]: Small suggestion, but also a general writing tip: `Areas can detect` -> `Areas detect`. I would remove the first sentence:
`General purpose area detection for 2D physics (see [CollisionObject2D] for usage). Areas can detect objects that enter or exit them.`
-> Full suggestion:
`2D area that detects nodes that enter or exit it. Change the 'space_override' property to AREA_SPACE_OVERRIDE_* to override physics parameters for nodes like [RigidBody2D]. E.g. gravity, damping... See [CollisionObject2D] for usage.`

File: doc/base/classes.xml [hunk 7]
```diff
@@ -5166,7 +5166,7 @@
 			<return type="Array">
 			</return>
 			<description>
-				Return a list of the areas that are totally or partially inside this area.
+				Return a list of the areas ([Area2D]) that are totally or partially inside this area.
```
Comment (NathanLovato) [inline doc/base/classes.xml]: Suggestion: `Returns a list of all the Area2Ds that intersect with this area`

**Review by NathanLovato — COMMENTED** [review]
Good job. Some fixes to do, + you can simplify some of the content. Please review the writing guidelines: avoid passive voice - https://github.com/GDquest/godot-3-guides/blob/master/contribution-guides/class-reference-styleguide.md

File: doc/base/classes.xml (hunk 4 above)
Comment (willnationsdev) [reply doc/base/classes.xml]: So the description should be on the same indent as the tag?

File: doc/base/classes.xml (hunk 6 above)
Comment (willnationsdev) [reply doc/base/classes.xml]: Done.

File: doc/base/classes.xml (hunk 7 above)
Comment (willnationsdev) [reply doc/base/classes.xml]: Done.

File: doc/base/classes.xml (hunk 5 above)
Comment (willnationsdev) [reply doc/base/classes.xml]: Done. Will start rebasing in a second.

**Comment by willnationsdev** [comment]
Well, I tried to do `git rebase -i HEAD~6`, `pick`ed the first, `squash`ed the remainnig 5, and got a new commit with everything together as one. But it seems to have only resulted in a merge and a re-submission of the super-commit, getting no change in content, but an additional 2 commits. :-/

File: doc/base/classes.xml (hunk 1 above)
Comment (willnationsdev) [reply doc/base/classes.xml]: I forgot to mention that I added all of these "if true" changes too.

File: doc/base/classes.xml (hunk 4 above)
Comment (NathanLovato) [reply doc/base/classes.xml]: No, it should be one more, and only tabs (2 spaces in your commit). You can see the original indent in red: 2 tabs

File: doc/base/classes.xml [hunk 8]
```diff
@@ -5083,10 +5083,10 @@
 </class>
 <class name="Area2D" inherits="CollisionObject2D" category="Core">
 	<brief_description>
-		General purpose area detection and influence for 2D physics.
+	2D area that detects nodes that enter or exit it. Can override 2D physics properties within range.
```
Comment (Noshyaar) [inline doc/base/classes.xml]: Should be one more tab.

**Comment by Noshyaar** [comment]
There are a few indentation issues. Can be merged if fixed and squashed.

File: doc/base/classes.xml (hunk 8 above)
Comment (willnationsdev) [reply doc/base/classes.xml]: I thought so. Someone told me to move it back, but I'll move it up again. XD

**Comment by willnationsdev** [comment]
Ok, I finally figured out how to get rebase working the way I want (just needed `git push -f` afterwards). I think things should all be good now.

**Comment by willnationsdev** [comment]
Well, then I tried to pull upstream after-the-fact, and that seems to have screwed things up again. *sigh* I already have the "finished" versions, so I'm thinking I'll just completely reset the branch, make sure it's pulled up to be in line with the current master, and then re-submit the actual change as a single commit. Give me some time. :-/

**Pull request merged**